Sources/OpenAI/Public/Models/AudioSpeechQuery.swift

```diff
@@ -43,36 +43,59 @@ public struct AudioSpeechQuery: Codable, Equatable {
     /// The voice to use when generating the audio. Supported voices are alloy, echo, fable, onyx, nova, and shimmer.
     public let voice: AudioSpeechVoice
     /// The format to audio in. Supported formats are mp3, opus, aac, and flac.
-    public let response_format: AudioSpeechResponseFormat
+    public let responseFormat: AudioSpeechResponseFormat
     /// The speed of the generated audio. Enter a value between **0.25** and **4.0**. Default: **1.0**
     public let speed: String?
     
+    public enum CodingKeys: String, CodingKey {
+        case model
+        case input
+        case voice
+        case responseFormat = "response_format"
+        case speed
+    }
+    
     public init(model: Model?,
                 input: String,
                 voice: AudioSpeechVoice,
-                response_format: AudioSpeechResponseFormat = .mp3,
+                responseFormat: AudioSpeechResponseFormat = .mp3,
                 speed: Double?) {
         
         self.model = {
-            guard let model else { return .tts_1 }
-            let isModelOfIncorrentFormat = model != .tts_1 && model != .tts_1_hd
-            guard !isModelOfIncorrentFormat else {
-                NSLog("[AudioSpeech] 'AudioSpeechQuery' must have a valid Text-To-Speech model, 'tts-1' or 'tts-1-hd'. Setting model to 'tts-1'.")
-                return .tts_1
-            }
-            return model
+            AudioSpeechQuery.validateSpeechModel(model)
```

Comment:
What is the main idea of validating model it inside closure? 

```swift
self.model = AudioSpeechQuery.validateSpeechModel(model)
```

Reply:
Oh, good point 🔥
Thanks)